python/ray/includes/unique_ids.pxi

```diff
@@ -106,40 +106,102 @@ cdef class UniqueID:
 
 
 cdef class ObjectID(UniqueID):
-    pass
+
+    def __init__(self, id):
+        if type(self) is ObjectID:
```

Comment:
Can this be invoked in a situation where self is not of type ObjectID?

Reply:
Maybe this is only useful in `__cinit__` and it is not necessary in `__init__`. I will do a double confirm. For `__cinit__`, the functions of derived class and base class will both be called, so using this `if` statement will avoid duplicated initiation.

Reply:
@pcmoritz In cython, `__init__` behaves the same as python's. This function will be called only once. On the contrast, `__cinit__` will be called in both base class and derived class. I have removed the type check. Thanks!